sympy/matrices/expressions/tests/test_derivatives.py

```diff
@@ -101,6 +101,9 @@ def test_matrix_derivative_vectors_and_scalars():
 
 def test_matrix_derivatives_of_traces():
 
+    expr = Trace(A)*A
+    assert expr.diff(A) == Trace(A)*Identity(k) + A.T
```

Comment:
@Upabjojr can you confirm if this is correct?

Reply:
```
Trace(A)*A) ===> A_{ii} A_{ij}
```
now
```
diff(A_ii A_jk, A_mn) = d_mi d_ni A_jk + A_ii d_mj d_nk = d_mn A_jk + A_ii d_mj d_nk
```
There is a problem with this result, as it should probably be a higher-rank array. MatrixCalculus.org is indeed unable to compute it.

Reply:
I'd say the result is wrong, it's omitting a higher dimensional identity.

Reply:
https://github.com/sympy/sympy/issues/15667